In [ ]:
---
title: "Lab 2 - Avocado Sales"
author: "Roee Morag"
date: "2024-10-07"
output: 
  html_document:
    theme: "cosmo" 
--- 


In [2]:
import pandas as pd
import numpy as np

In [25]:
avocado_df = pd.read_csv("C:\\Users\\morag\\OneDrive\\Documents\\MSBA\\GSB 544\\Labs\\Labs---GSB-544\\Data\\avocado-updated-2020.csv")

In [26]:
avocado_df.head(n=4)

date  average_price  total_volume       4046  ...  xlarge_bags          type  year  geography
0  2015-01-04           1.22      40873.28    2819.50  ...          0.0  conventional  2015     Albany
1  2015-01-04           1.79       1373.95      57.42  ...          0.0       organic  2015     Albany
2  2015-01-04           1.00     435021.49  364302.39  ...          0.0  conventional  2015    Atlanta
3  2015-01-04           1.76       3846.69    1500.15  ...          0.0       organic  2015    Atlanta

[4 rows x 13 columns]

1. This data set is about avocado prices in different regions/cities of the United States. Each row/observation represents a day of avocado sales in a specific city, i.e Albany, NY or Atlanta, GA. Each region/city has two observations for each date though, one showing their sales of conventional avocados and the other showing their sales of organic avocados. Each of thesscities is also a subset of a region which is a subset of the total United States column. Columns in the data set include average price of the avocados sold, the total volume of avocados sold, total bags, small bags, large bags, and extra large bags sold. The 4046, 4225, and 4770 columns are the price look up codes for smaller, medium, and large avocados and the columns refer to the amount sold on the given day.

2. As far as cleaning this data set goes, I will clean it as I go through the problems - not quite sure how to organize this until I have intiative with the problems. I will rename some columns, such as 4046, to more intuitive naming schemes.

In [27]:
avocado_df.rename(columns={"4046": "small_avocados", "4225": "large_avocados", "4770": "xlarge_avocados"}, inplace=True)

3. Which major geographical region sold the most total organic, small Hass avocados in 2017?

For this question, I will filter a data set to include only the year 2017 and only organic av ocado sales. Then, I will group by region, sum along the column of small_avocados and sort by that column descending. I do not know which major geographical regions are covered in this dataset iwhtout combing through the 700ish entries but this method should give me a good idea. 

In [28]:
organic_2017 = avocado_df[(avocado_df["type"] == "organic") & (avocado_df["year"] == 2017)]

In [37]:
small_haas_2017_sum = organic_2017.groupby("geography")["small_avocados"].sum().reset_index().sort_values(by="small_avocados", ascending=False)

small_haas_2017_sum.head(n=15)

geography  small_avocados
51        Total U.S.      6882024.42
52              West      1870206.29
45     South Central      1717104.19
6         California      1526290.58
29         Northeast       925832.36
22       Los Angeles       600718.66
28          New York       599671.34
12            Denver       556346.10
18           Houston       498024.65
42     San Francisco       458478.55
11  Dallas/Ft. Worth       373197.18
46         Southeast       285124.34
15       Great Lakes       202487.68
25          Midsouth       186395.50
33    Phoenix/Tucson       184084.38

I am going to assume that there are five major geographical regions (as I'm seeing in the data). They are West, South Central, Northeast, Southeast, and Midsouth. The West region sold the most total organic, small Haas avocados in 2017 with 1870206.29.

4. Split the date variable into month, day, and year variables. In which month is the highest average volume of avocado sales?

This is going to be a similar process to the last problem. First, we will create new variables for day and month (unnecessary for year as we already have a variable for that). Then we will filter out the data set to only include the geography of "Total US". Finally, we will group by the month and average out the total volume of avocados sold.

In [41]:
avocado_df["date"] = pd.to_datetime(avocado_df["date"])

avocado_df["month"] = avocado_df["date"].dt.month
avocado_df["day"] = avocado_df["date"].dt.day

In [42]:
total_us = avocado_df[avocado_df["geography"] == "Total U.S."]

In [46]:
total_us.groupby("month")["total_volume"].mean().reset_index().sort_values(by="total_volume", ascending=False)

month  total_volume
4       5  2.293203e+07
1       2  2.236790e+07
5       6  2.144947e+07
3       4  2.077160e+07
2       3  2.035640e+07
0       1  2.011432e+07
6       7  2.009696e+07
7       8  1.938566e+07
8       9  1.847439e+07
9      10  1.719583e+07
10     11  1.673651e+07
11     12  1.645596e+07

The month of May sees the highest average volume of avocado sales.